File: python/sparklabs/batch/sales/top_selling_products.py
from pyspark.sql import functions as F

from sparklabs.batch.sales import utils


def run() -> None:
    (
        utils.order_items.join(utils.products, ["product_id"], "inner")
        .groupBy(F.col("product_id"))
        .agg(F.count("*").alias("total_orders"))
        .orderBy(F.desc("total_orders"))
        .show(truncate=False)
    )


if __name__ == "__main__":
    run()

File: r/sparklabs/batch/sales/dataframe/top_selling_products.R
source("r/sparklabs/batch/sales/dataframe/utils.R")

result <- orderBy(
  agg(groupBy(order_items, order_items$product_id), total_orders = count(order_items$product_id)),
  desc("total_orders")
)
showDF(result, truncate = FALSE)
